### ISSUE TITLE
Popover Component

### ISSUE THREAD

**Issue by leventozen** [Trendyol/baklava#323]
TBD

**Comment by buseselvi** [comment]
[Design Documentation](https://www.figma.com/file/RrcLH0mWpIUy4vwuTlDeKN/Baklava-Design-Guide?node-id=3369%3A26076&t=9atyD6p1ZMzRgKUl-1)

**Comment by muratcorlu** [comment]
Good job @ozkersemih! Here are my points:

* `target` property can accept a direct reference of the target dom element as well as just id of it. So in frameworks we can take advantage of setting `target` property with a reference of an element without needing to give a uniquie id to it.
* As a principle, it would be great to accept all of the "visual" parameters as CSS variables. For me `arrow`, `offset`, `inline`, `shift` and `fitSize` are all visual parameters (doesn't change functionality). I know sometimes it's technically very difficult to take them as CSS variables since they are used in JS code as parameters of Floating UI, but here it would be great to be sure that: they are really needed and they need to be set as attributes instead of CSS variables. I can also check for the opportunities to use them as CSS variables but for arrow, I'm pretty sure we can take a `--bl-popover-arrow` as the `display` value of arrow element inside. So if developer doesn't want to have arrow, they can set it as `--bl-popover-arrow: none;`.
* I think should have some public methods like `show()` and `hide()`
* Popover can be triggered by a click on target element and with a `mouseenter` as well (for tooltip). I think we need extra arguments to define those events (with some nice defaults).
* In some use cases we want to sync width of popover with it's trigger element (select, dropdown). I think we'll need a parameter for enabling this.

**Comment by ozkersemih** [comment]
Hi @muratcorlu thanks for your review.
* We can accept target as HTML element, its okay.
* We will try accept visual parameters with CSS variables as possible as.
* I think we don't need public `show` and `hide` method. We can already have `open` attribute and we can trigger `show` and `hide` methods with this attribute.
* We think popover is utility component. So maybe it doesn't need any attribute about `mouseenter` or `click`. 
  * e.g we can handle `mouseenter` and `mouseleave` in `tooltip` component. These methods can trigger popovers open attribute.
* We defined parameter `fitSize` for that.

**Comment by muratcorlu** [comment]
> * We think popover is utility component. So maybe it doesn't need any attribute about `mouseenter` or `click`.
>   
>   * e.g we can handle `mouseenter` and `mouseleave` in `tooltip` component. These methods can trigger popovers open attribute.

So popover will never open itself by listening some events of the trigger element then?

> * We defined parameter `fitSize` for that.

Then I'm not sure if `fitSize` really gives that feeling. And description should mention this like "Enabling this will sync width of the popover with the trigger element's width".

**Comment by ozkersemih** [comment]
> So popover will never open itself by listening some events of the trigger element then?
You are right, we will use public methods. So should we add these methods to `ADR` and `storybook`? I see same methods exists in `tooltip` component but I don't see methods on storybook of `tooltip`.

**Comment by muratcorlu** [comment]
Unfortunately Storybook reference table doesn't automatically include public methods. We already have an issue for that but no solution for now: #221 But this doesn't block us to mention those in our ADR document.

But to be honest, if popover would open and close itself in some common -and customizable- use-cases, it would be helpful I think. Otherwise, we'll do the same thing in every internal component to close the popover when clicked outside of it.

**Comment by muratcorlu** [comment]
* Please indicate that `target` can be either a `string` as an element ID or a reference of an `HTMLElement` as the target element.
* Use dash case for attribute (`fit-size`)
* What about styling popover wrapper? If we'll have some extra CSS variables for them (like padding, border) please add them as well.

**Comment by ozkersemih** [comment]
@muratcorlu Hi, I have updated adr with your suggestions
